### PR TITLE
Es 5 support

### PR DESCRIPTION
**Pull request by johnspicer**
@EDITD/hackers - This updates the ES role for ES v5.

The only differences it needs to know about are the different url for downloading the deb, and not installing plugins.

There is also a change here that handles support for the openjdk version difference between Ubuntu 14.04 and 16.04!

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@
   shell: /usr/share/elasticsearch/bin/plugin {% if elasticsearch_file_name|get_major_version == 1 %}--{% endif %}install {{ item }}
   with_items: "{{ plugins_to_install }}"
   register: plugin_response
+  when: elasticsearch_file_name|get_major_version < 5
```
Comment (pkatseas) [inline tasks/main.yml:27]: If I'm not mistaken, some plugins could be installed with ES5, right?
It's just site plugins that are no longer allowed I think.

File: tasks/main.yml [hunk 2]
```diff
@@ -1,8 +1,7 @@
 ---
 
 - name: install ES dependencies
-  apt: pkg={{ item }} state=present update_cache=true cache_valid_time=3600
-  with_items: [openjdk-7-jre-headless]
+  apt: pkg=openjdk-{{ ansible_distribution_version|get_openjdk_version }}-jre-headless state=present update_cache=true cache_valid_time=3600
```
Comment (pkatseas) [inline tasks/main.yml]: I'm pretty sure you can use 'placeholder' packages like `default-jre-headless` in this instance, to install cross-version packages. I remember using that for ES purposes before.

For example on 16.04:
```
$ apt-cache show default-jre-headless
Package: default-jre-headless
Priority: optional
Section: interpreters
Installed-Size: 22
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian Java Maintainers <pkg-java-maintainers@lists.alioth.debian.org>
Architecture: amd64
Source: java-common (0.56ubuntu2)
Version: 2:1.8-56ubuntu2
Replaces: openjdk-6-jre-headless (<< 6b23~pre8-2~)
Provides: java-runtime-headless, java2-runtime-headless, java5-runtime-headless, java6-runtime-headless, java7-runtime-headless, java8-runtime-headless
Depends: openjdk-8-jre-headless, java-common
Suggests: default-jre
Filename: pool/main/j/java-common/default-jre-headless_1.8-56ubuntu2_amd64.deb
Size: 4380
MD5sum: 524b8dfba00019adf231d9951cf2a7ac
SHA1: 3459b0366ba3c09ef5b3ee7fe3577da90533a6fc
SHA256: b16b4b116a2533a85f766a09d6182adb8a8ac975148c143c4c2ed8986991d44d
Description-en: Standard Java or Java compatible Runtime (headless)
 This package points to the Java runtime, or Java compatible
 runtime recommended for this architecture, which is
 openjdk-8-jre-headless for amd64.
 .
 The package is used as dependency for packages not needing a
 graphical display during runtime.
Description-md5: 37ab6c1486c2ec477fc03127f0b1995e
Homepage: https://wiki.debian.org/Java/
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Origin: Ubuntu
Supported: 5y
Task: ubuntu-usb, tomcat-server, kubuntu-desktop, kubuntu-full, edubuntu-usb, ubuntu-mate-desktop
```

And on 14.04:
```
$ apt-cache show default-jre-headless
Package: default-jre-headless
Priority: optional
Section: interpreters
Installed-Size: 48
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Debian Java Mailing List <debian-java@lists.debian.org>
Architecture: amd64
Source: java-common (0.51)
Version: 2:1.7-51
Replaces: openjdk-6-jre-headless (<< 6b23~pre8-2~)
Provides: java-runtime-headless, java2-runtime-headless, java5-runtime-headless, java6-runtime-headless, java7-runtime-headless
Depends: openjdk-7-jre-headless (>= 7~u3-2.1.1), java-common
Suggests: default-jre
Filename: pool/main/j/java-common/default-jre-headless_1.7-51_amd64.deb
Size: 3834
MD5sum: 82babba8748dca94fddb6449c2e0c38e
SHA1: ee5324aacd2f4a1502a222957a55e3e2453c0409
SHA256: df0d661d8dc83a066a40e9a49fa39a0cec7f2fb1e69775536f8a7f6a1863cf4e
Description-en: Standard Java or Java compatible Runtime (headless)
 This package points to the Java runtime, or Java compatible
 runtime recommended for this architecture, which is
 openjdk-7-jre-headless for amd64.
 .
 The package is used as dependency for packages not needing a
 graphical display during runtime.
Description-md5: e5da5cbbcd2bc8982f52a7a482d2a0d4
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Origin: Ubuntu
Supported: 5y
Task: ubuntu-usb, tomcat-server, kubuntu-desktop, kubuntu-full, kubuntu-active-desktop, kubuntu-active-full, edubuntu-usb
```

Key field being the `Depends` one.

**Review by pkatseas — CHANGES_REQUESTED** [review]
LGTM @johnspicer :+1: 
I've added a couple of suggestions

File: tasks/main.yml (hunk 2 above)
Comment (johnspicer) [reply tasks/main.yml]: Ah thats brilliant! I didn't like having to add another function just for that. I'll update now, thanks @pkatseas !

Comment (johnspicer) [reply tasks/main.yml]: https://github.com/EDITD/ansible-elasticsearch_dev/pull/4/commits/1f95596bb11cc49b4baa5f3a00cad709fbb46182 👍

File: tasks/main.yml (hunk 1 above)
Comment (johnspicer) [reply tasks/main.yml:27]: All of the plugins we use are site plugins though right? The only things mentioned in the docs are site plugins, or installing plugins [using kibana](https://www.elastic.co/guide/en/kibana/current/_installing_plugins.html)

**Review by pkatseas — APPROVED** [review]

**Comment by johnspicer** [comment]
Thanks for the review @pkatseas!

**Pull request merged**